src/doc/tarpl/unchecked-uninit.md

```diff
@@ -0,0 +1,85 @@
+% Unchecked Uninitialized Memory
+
+One interesting exception to this rule is working with arrays. Safe Rust doesn't
+permit you to partially initialize an array. When you initialize an array, you
+can either set every value to the same thing with `let x = [val; N]`, or you can
+specify each member individually with `let x = [val1, val2, val3]`.
+Unfortunately this is pretty rigid, especially if you need to initialize your
+array in a more incremental or dynamic way.
+
+Unsafe Rust gives us a powerful tool to handle this problem:
+`mem::uninitialized`. This function pretends to return a value when really
+it does nothing at all. Using it, we can convince Rust that we have initialized
+a variable, allowing us to do trickier things with conditional and incremental
+initialization.
+
+Unfortunately, this opens us up to all kinds of problems. Assignment has a
+different meaning to Rust based on whether it believes that a variable is
+initialized or not. If it's uninitialized, then Rust will semantically just
+memcopy the bits over the uninitialized ones, and do nothing else. However if Rust
+believes a value to be initialized, it will try to `Drop` the old value!
+Since we've tricked Rust into believing that the value is initialized, we
+can no longer safely use normal assignment.
```

Comment:
Nit: well, we could for types that implement `Copy` since those cannot implement `Drop`.

(I suppose ideally we'd have some way of saying `!Drop` in general. But, we're not there yet.)

((Oh, you say this below. Hmm, okay, maybe it doesn't have to be stated up front here, I don't know ...))